File: protoqueue-common/src/main/java/net/protoqueue/annotation/ProtoQueueClass.java
package net.protoqueue.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 注解在入口类上
 * Created by zhongyongsheng on 2018/4/20.
 */
@Retention(RetentionPolicy.CLASS)
@Target(ElementType.TYPE)
public @interface ProtoQueueClass {

    /**
     * seq上下文在proto里定义的名称
     * @return
     */
    String protoContextLiteral();

    /**
     * 生成buildProto的语句，默认例如FtsUserProto.parseFrom
     * @return
     */
    String buildProtoLiteral() default "return %T.parseFrom(%L)";

    /**
     * 生成toByteArray的语句
     * @return
     */
    String toByteArrayLiteral() default "return com.google.protobuf.nano.MessageNano.toByteArray(proto)";

    /**
     * uri在proto里定义的名称
     * @return
     */
    String uriLiteral() default "uri";
}

File: protoqueue-common/src/main/java/net/protoqueue/annotation/ProtoQueueClass.kt
package net.protoqueue.annotation

/**
 * 注解在入口类上
 * Created by zhongyongsheng on 2020/4/20.
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class ProtoQueueClass(
    /**
     * seq上下文在proto里定义的名称
     * %L传入的是proto
     * @return
     */
    val protoContextLiteral: String = "return %L.header?.seqid",
    /**
     * 生成buildProto的语句，默认例如FtsUserProto.parseFrom
     * %T是PROTO类名
     * %L是data
     * @return
     */
    val buildProtoLiteral: String = "return %T.parseFrom(%L)",
    /**
     * 生成toByteArray的语句
     * @return
     */
    val toByteArrayLiteral: String = "return com.google.protobuf.nano.MessageNano.toByteArray(proto)",
    /**
     * uri在proto里定义的名称
     * @return
     */
    val uriLiteral: String = "uri",

    /**
     * rpc方法中使用到的获取response header代码
     * %L是PROTO的实例
     */
    val resHeaderLiteral: String = "%L.header"
)